mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,169 @@
+        """
+        after mlrun function's run is done, copy all data logged by third party app tracker
+        :param context: current mlrun context
+        :param run: mlflow run to log
+        """
+        model_path = []
+        for key, val in run.data.params.items():
+            context._parameters[key] = val
+        context.log_results(run.data.metrics)
+        context.set_label("mlflow-runid", run.info.run_id)
+        context.set_label("mlflow-experiment", run.info.experiment_id)
+        for artifact in self._client.list_artifacts(run.info.run_id):
+            full_path = self._tracked_platform.artifacts.download_artifacts(
+                run_id=run.info.run_id, artifact_path=artifact.path
+            )
+            if artifact.is_dir and os.path.exists(os.path.join(full_path, "MLmodel")):
+                model_path.append(
+                    full_path
+                )  # this is the model folder, we log it after logging all artifacts
+            else:
+                self.log_artifact(context, full_path, artifact)
+        if model_path:
+            for path in model_path:
+                self.log_model(path, context)
+
+    def log_model(
+        self,
+        model_uri: str,
+        context: Union[MLClientCtx, dict],
+    ):
+
+        model_info = self._tracked_platform.models.get_model_info(model_uri=model_uri)
+        tmp_dir = tempfile.TemporaryDirectory()
+        model_zip = f"{tmp_dir.name}model.zip"
+        zip_folder(model_uri, model_zip)
+        key = model_info.artifact_path
+        inputs = outputs = None
+
+        if model_info.signature is not None:
+            if model_info.signature.inputs is not None:
+                inputs = schema_to_feature(
+                    model_info.signature.inputs, convert_np_dtype_to_value_type()
+                )
+            if model_info.signature.outputs is not None:
+                outputs = schema_to_feature(
+                    model_info.signature.outputs, convert_np_dtype_to_value_type()
+                )
+        try:
+            context.log_model(
+                key,
+                framework="mlflow",
+                model_file=model_zip,
+                metrics=context.results,
+                parameters=model_info.flavors,
+                labels={
+                    "mlflow_run_id": model_info.run_id,
+                    "mlflow_version": model_info.mlflow_version,
+                    "model_uuid": model_info.model_uuid,
+                },
+                extra_data=self._artifacts,
+                inputs=inputs,
+                outputs=outputs,
+            )
+            tmp_dir.cleanup()
+        except Exception:
+            tmp_dir.cleanup()
+
+    def log_artifact(self, context, full_path, artifact):
+        artifact = context.log_artifact(
+            item=pathlib.Path(artifact.path).name.replace(".", "_"),
+            local_path=full_path,
+        )
+        self._artifacts[artifact.key] = artifact
```

Comment:
why can't it be moved to `BaseTracker`? there is nothing related to MLFlow here, and it can reduce code duplications when implementing new trackers.